src/test/java/catan/controllers/testcases/game/FinishGameTest.java

```diff
@@ -143,4 +94,123 @@ public void should_successfully_finish_game_when_target_victory_points_is_4_and_
+                .setHex(HexType.BRICK, 6).atCoordinates(0, 2)
+
+                .createNewPublicGameByUser(USER_NAME_1, targetVictoryPoints)
+                .joinPublicGame(USER_NAME_2)
+                .joinPublicGame(USER_NAME_3)
+
+                        // take last player from the list each time, when pulling move order from the list to have order: 3, 2, 1
+                .nextRandomMoveOrderValues(asList(3, 2, 1))
+
+                .setUserReady(USER_NAME_1)
+                .setUserReady(USER_NAME_2)
+                .setUserReady(USER_NAME_3);
+    }
+
+    private Scenario playPreparationStage() {
+        return scenario
+                .BUILD_SETTLEMENT(1).atNode(2, -2, "topLeft")
+                .BUILD_ROAD(1).atEdge(2, -2, "topLeft")
+                .END_TURN(1)
+
+                .BUILD_SETTLEMENT(2).atNode(2, -1, "bottomRight")
+                .BUILD_ROAD(2).atEdge(2, -1, "bottomRight")
+                .END_TURN(2)
+
+                .BUILD_SETTLEMENT(3).atNode(0, 2, "topRight")
+                .BUILD_ROAD(3).atEdge(0, 2, "topRight")
+                .END_TURN(3)
+
+                .BUILD_SETTLEMENT(3).atNode(0, 0, "bottomRight")
+                .BUILD_ROAD(3).atEdge(0, 0, "bottomRight")
+                .END_TURN(3)
+
+                .BUILD_SETTLEMENT(2).atNode(0, 0, "bottomLeft")
+                .BUILD_ROAD(2).atEdge(0, 0, "bottomLeft")
+                .END_TURN(2)
+
+                .BUILD_SETTLEMENT(1).atNode(0, 0, "top")
+                .BUILD_ROAD(1).atEdge(0, 0, "topLeft")
+                .END_TURN(1);
+    }
+
+    private Scenario giveResourcesToPlayerForRoadBuilding(int moveOrder) {
+        return scenario
+                .nextRandomDiceValues(asList(moveOrder, moveOrder, 6, 6, 6, 6))
+                .THROW_DICE(moveOrder)
+                .END_TURN(moveOrder)
+                .THROW_DICE(moveOrder == 1 ? 2 : moveOrder == 2 ? 3 : 1)
+                .END_TURN(moveOrder == 1 ? 2 : moveOrder == 2 ? 3 : 1)
+                .THROW_DICE(moveOrder == 1 ? 3 : moveOrder == 2 ? 1 : 2)
+                .END_TURN(moveOrder == 1 ? 3 : moveOrder == 2 ? 1 : 2);
+    }
+
+    private Scenario giveResourcesToPlayerForSettlementBuilding(int moveOrder) {
+        return scenario
+                .nextRandomDiceValues(asList(moveOrder, moveOrder, moveOrder, moveOrder == 3 ? moveOrder + 2 : moveOrder + 1, 6, 6))
+                .THROW_DICE(moveOrder)
+                .END_TURN(moveOrder)
+                .THROW_DICE(moveOrder == 1 ? 2 : moveOrder == 2 ? 3 : 1)
+                .END_TURN(moveOrder == 1 ? 2 : moveOrder == 2 ? 3 : 1)
+                .THROW_DICE(moveOrder == 1 ? 3 : moveOrder == 2 ? 1 : 2)
+                .END_TURN(moveOrder == 1 ? 3 : moveOrder == 2 ? 1 : 2);
+    }
+
+    /*
+    *          (X, Y) coordinates of generated map:                          Node position at hex:
+    *
+    *           *----*----*----*----*----*----*                                      top
```

Comment:
this map was drown just below the playPreparationStage() method to keep in view where I should build settlements and roads, why do we need it below giveResourcesToPlayerForSettlementBuilding() ?